libcontainer/container_linux.go

```diff
@@ -1124,6 +1144,17 @@ func (c *linuxContainer) runType() (Status, error) {
 		}
 		return Stopped, newSystemErrorWithCausef(err, "sending signal 0 to pid %d", pid)
 	}
+	// check if the init process is still the same process as the initial one,
+	// it could happen that the original process has exited and a new process
+	// has been created with the same pid, in this case, the container would
+	// already be stopped.
+	u, err := c.isProcessUnique(pid)
+	if err != nil {
+		return Stopped, err
+	}
+	if !u {
+		return Stopped, nil
```

Comment:
This is pretty much the same as the previous conditional, so you can collapse to:

    if !u || err != nil {
      return Stopped, err
    }

although if we rename `isProcessUnique` to `doesProcessExist` we probably want to rename `u` to `exists`.